bin/pry

```diff
@@ -17,6 +19,10 @@ def udec(*)
   UBID.decode(*)
 end
 
+def udec_deleted(ubid)
+  DeletedRecord.where(Sequel.pg_json(:model_values)["name"] => Sequel.pg_jsonb_wrap(UBID.parse(ubid).to_uuid)).first
```

Comment:
I think we must add a gin index for this field. Simply querying deleted records in production is almost unbearable at this point.

Reply:
it's also using almost all our space. Should we consider partitioning as well? How often are we digging in the ancient past? Not often, right?

GIN is perhaps the first thing to try, though the idea of a GIN over all (ever growing) history that has to get updated with every deletion makes me a little bit suspenseful. Just a bit.